packages/backend/src/apps/gathersg/actions/get-case-details/get-data-out-metadata.ts

```diff
@@ -0,0 +1,127 @@
+import { IDataOutMetadata, IExecutionStep, IJSONArray } from '@plumber/types'
+
+import { dataOutSchema } from './schema'
+
+async function getDataOutMetadata(
+  step: IExecutionStep,
+): Promise<IDataOutMetadata> {
+  const { dataOut: rawDataOut } = step
+  if (!rawDataOut) {
+    return null
+  }
+
+  const parsedDataOut = dataOutSchema.safeParse(rawDataOut)
+  if (!parsedDataOut.success) {
+    return null
+  }
+
+  const { data: dataOut } = parsedDataOut.data
+
+  const metadata = {
+    data: {
+      type: {
+        name: { label: 'Case type' },
+        uuid: { isHidden: true },
+      },
+      uuid: { label: 'Case UUID' },
+      caseRef: { label: 'Case ref' },
+      source: { label: 'Case source' },
+      status: {
+        name: { label: 'Status' },
+        uuid: { isHidden: true },
+        color: { isHidden: true },
+        isFinal: { isHidden: true },
+      },
+      createdAt: { label: 'Created at' },
+      createdBy: {
+        name: { label: 'Created by (name)' },
+        email: { label: 'Created by (email)' },
+        role: { isHidden: true },
+        uuid: { isHidden: true },
+      },
+      updatedAt: { label: 'Updated at' },
+      updatedBy: {
+        name: { label: 'Updated by (name)' },
+        email: { label: 'Updated by (email)' },
+        role: { isHidden: true },
+        uuid: { isHidden: true },
+      },
+      durationSec: { isHidden: true },
+      traceId: { label: 'Trace ID' },
+    },
+  }
+
+  // handle tags if any
+  // tags are an array of strings and exist at the top level alongside the
+  // uuid, caseRef, etc
+  const tagsMetadata = Object.create(null)
+  if (dataOut.tags && Array.isArray(dataOut.tags)) {
+    for (let i = 0; i < dataOut.tags.length; i++) {
+      tagsMetadata[i] = { label: `Tag` }
```

Comment:
nit: `tagsMetadata[i] = { label: `Tag ${i + 1}` }` so at least the tags look different from one another? Tbh idk how would that help much also cus tag 1, tag 2 a bit not useful also hmm